webapp/src/components/sidebar_right/index.tsx

```diff
@@ -0,0 +1,257 @@
+    org: string;
+    gitlabURL: string;
+    reviews: Item[];
+    unreads: Item[],
+    yourPrs: Item[],
+    yourAssignments: Item[],
+    rhsState: string,
+    theme: Theme,
+}
+
+export function renderView(props: Props) {
+    return (
+        <div
+            {...props}
+            className='scrollbar--view'
+        />);
+}
+
+export function renderThumbHorizontal(props: Props) {
+    return (
+        <div
+            {...props}
+            className='scrollbar--horizontal'
+        />);
+}
+
+export function renderThumbVertical(props: Props) {
+    return (
+        <div
+            {...props}
+            className='scrollbar--vertical'
+        />);
+}
+
+function mapGitlabItemListToPrList(gilist: Item[]) {
+    if (!gilist) {
+        return [];
+    }
+
+    return gilist.map((pr: Item) => {
+        return {sha: pr.sha, project_id: pr.project_id, iid: pr.iid};
+    });
+}
+
+function shouldUpdateDetails(prs: Item[], prevPrs: Item[], targetState: string, currentState: string, prevState: string) {
+    if (currentState !== targetState) {
+        return false;
+    }
+
+    if (currentState !== prevState) {
+        return true;
+    }
+
+    if (prs.length !== prevPrs.length) {
+        return true;
+    }
+
+    for (let i = 0; i < prs.length; i++) {
+        if (prs[i].id !== prevPrs[i].id) {
+            return true;
+        }
+    }
+    return false;
+}
+
+function mapPrsToDetails(prs: Item[], details: Item[]) {
+    if (!prs || !prs.length) {
+        return [];
+    }
+
+    return prs.map((pr) => {
+        let foundDetails;
+        if (details) {
+            foundDetails = details.find((prDetails) => {
+                return (pr.project_id === prDetails.project_id) && (pr.sha === prDetails.sha);
+            });
+        }
+        if (!foundDetails) {
+            return pr;
+        }
+
+        return {
+            ...pr,
+            status: foundDetails.status,
+            approvers: foundDetails.approvers,
+            total_reviewers: pr.reviewers.length,
+        };
+    });
+}
+
+function SidebarRight({theme}: {theme: Theme}) {
+    const {username, yourAssignments, org, unreads, gitlabURL, rhsState, reviews, yourPrs, reviewDetails, yourPrDetails} = useSelector((state) => {
+        const store = getPluginState(state);
+        return {
+            username: store.username,
+            reviews: store.reviews,
+            reviewDetails: store.reviewDetails,
+            yourPrs: store.yourPrs,
+            yourPrDetails: store.yourPrDetails,
+            yourAssignments: store.yourAssignments,
+            unreads: store.unreads,
+            org: store.organization,
+            gitlabURL: store.gitlabURL,
+            rhsState: store.rhsState,
+        };
+    });
+
+    const [updatedPrs, setUpdatedPrs] = useState<Item[]>(yourPrs);
+    const [RHSState, setRHSState] = useState('');
+    const [updatedReviews, setUpdatedReviews] = useState<Item[]>(reviews);
+    const dispatch = useDispatch();
```

Comment:
There were no state variables previously in this component when it was a class component. What's the need for these state variables here?

Reply:
we need to compare the new state with the previous state, so using the useState to store previous states

Reply:
Sync with Raghav. He created a `usePrevious` hook in a PR, see if that can be used here. @raghavaggarwal2308 @ayusht2810

Reply:
these states store the PRs and also add new values like (status, approvers, total_reviewers) from the function "mapPrsToDetails"